File: app/src/main/java/com/example/braincard/data/model/Card.kt
package com.example.braincard.data.model

import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.PrimaryKey

@Entity(tableName = "Card", foreignKeys = [ ForeignKey(entity = Deck::class, parentColumns = ["id"], childColumns = ["deckID"], onDelete = ForeignKey.CASCADE)])
//anche Entity è una annotation
data class Card (
    @PrimaryKey val id : String = "",// questa è una annotation
    var domanda : String = "",
    var risposta : String = "",
    var completata : Boolean = false,
    var deckID : String
)

File: FLUTTER/braincard_flutter/lib/model/card.dart
class Card {
  final String id;
  final String domanda;
  final String risposta;
  int completata;
  final String deckID;

  Card({
    required this.id,
    required this.domanda,
    required this.risposta,
    required this.completata,
    required this.deckID,
  });

  Map<String, dynamic> toMap() {
    return {
      'id': id,
      'domanda': domanda,
      'risposta': risposta,
      'completata': completata, 
      'deckID': deckID,
    };
  }

  factory Card.fromMap(Map<String, dynamic> map) {
    return Card(
      id: map['id'],
      domanda: map['domanda'],
      risposta: map['risposta'],
      completata: map['completata'], 
      deckID: map['deckID'],
    );
  }
}
